### ISSUE TITLE
DragonProxy Plugins

### ISSUE THREAD

**Issue by File14** [DragonetMC/DragonProxy#152]
Hello developers!

Will there ever be a way to code plugins for DragonProxy? As you do do with Bungeecord.

Global commands that might be useful. Such as: /Find, /Send, /Vote, /Store, /Hub, /Glist or /list and /Kick.

This is my suggestion, thanks for now! :)

Best Regards,
File14

**Comment by TekkitCommando** [comment]
@DefinitlyEvil What do you think? :P

**Comment by TheDiamondYT1** [comment]
You could just add the commands to the PC server, but if you want it to be client side then we may add addonnsupport in the future.

**Comment by Tee7even** [comment]
@TheDiamondYT1 consider this: you have a bunch of servers to which players connect using a proxy (i.e. server network). It would be nice if they had to register once for the whole network, not for every server, right? And also hubs, minigames (I suppose) and a lot of other stuff. That's why proxy plugins.

**Comment by DefinitlyEvil** [comment]
Plugins are not useful, since u can point DragonProxy to a BungeeCord server which has plugin support.

**Comment by File14** [comment]
Wait, does MCPE supports Bungeecord? O.o I don't get it.

**Comment by Pet0203** [comment]
Yes, you can connect DragonProxy to a Bungeecord network.

**Comment by File14** [comment]
Yes, but will MCPE clients be allowed to connect through Bungeecord?

**Comment by Pet0203** [comment]
They are allowed if you connect through DragonProxy, to Bungeecord. Quote from DragonProxy readme: 
"Supported Remote Server Types

For PC/Mac Servers: Bukkit/Spigot/Glowstone/SpongeVanilla/**BungeeCord**
For Pocket Edition Servers: MiNet, Nukkit, PocketMine-MP (and all forks)
Not listed is NOT mean unsupported, just untested."

**Comment by File14** [comment]
> Spigot server
So I can do: DragonProxy > Bungeecord ->
                                                                     > MCPE server 

Can I use MCPE and Spigot servers? So players can /Server between them?

**Comment by Pet0203** [comment]
Oh, you mean like that. No, Bungeecord does not support MCPE servers like PocketMine, Nukkit. I think that you will be able to switch between them with some sort of command. 

@DefinitlyEvil I think we'll have to add plugin support, because you cant point Bungeecord to a MCPE server.

**Comment by TheDiamondYT1** [comment]
It would be called "addons. I think we should add it because you could add custom proxy commands,  you could display custom messages on joining the proxy, you could send stats to your website, etc

**Comment by Pet0203** [comment]
But that would be later, because I think we need to focus on **0.15.0** and **1.10** first..

**Comment by TheDiamondYT1** [comment]
Yes, and we may keep it 1.8 for a while, due to many servers not updating yet.

**Comment by File14** [comment]
Yes, you are right. Many uses ViaVersion to hav 1.8.-1.10 support.

I think you guys should add MCPE 0.15.0 support first of all. :)

**Comment by TheDiamondYT1** [comment]
Many use ProtocolLib or ProtocolSupport whatever the called

**Comment by File14** [comment]
Yes but if you want to use Protocolsupport then you must update your server to 1.10 to be able to support all versions down to 1.8. 

So more people are using Viaversion because then you don't need to update your server and mess with the plugins.

**Comment by Aericio** [comment]
@Tee7even Can't you just use MySQL or something and link to the auth plugin your using? or is that something else

**Comment by Tee7even** [comment]
@Aericio of course,  but I don't think it's a real solution, it's kind of a workaround. Anyway, there's much more use for proxy plugins than this, if DragonProxy devs don't want to add them and think that PC redirect is a cool solution for all this, fine, it's their business.

**Comment by TheDiamondYT1** [comment]
Addons WILL be a thing. I think I might make it java addons not javascript, because we dont need to add as much code haha.

**Comment by DefinitlyEvil** [comment]
Hmm, we should think about making an API. xD

**Comment by TheDiamondYT1** [comment]
Yep

**Comment by TheDiamondYT1** [comment]
Being worked on in branch **API**.

**Comment by TheDiamondYT1** [comment]
@DefinitlyEvil How about making a submodule called "api" and then rename the current module to "proxy".

**Comment by TheDiamondYT1** [comment]
Then some of the class can implement classes from the API.

**Comment by TheDiamondYT1** [comment]
@DefinitlyEvil Anyway, 0.15 first - API later. It shouldnt take you long to implement 0.15 look at nukkit.
